### ISSUE TITLE
🐛 [firebase_core] firestore offline don't work on flutter web 

### ISSUE THREAD

**Issue by YawarOsman** [firebase/flutterfire#10259]
## Bug report

**Describe the bug**
I have tried to enable Persistence in my `flutter web` application as i provided the code down below, but it work and shows this error:

``` 
import 'package:cloud_firestore/cloud_firestore.dart';
import 'package:firebase_core/firebase_core.dart';
import 'package:flutter/material.dart';
import 'package:flutterfire/test.dart';

import 'firebase_options.dart';

void main() async {
  WidgetsFlutterBinding.ensureInitialized();

  await Firebase.initializeApp(options: DefaultFirebaseOptions.currentPlatform);
  try {
    FirebaseFirestore.instance
        .enablePersistence(const PersistenceSettings(synchronizeTabs: true));
  } catch (e) {
    print(e);
  }
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: const Test(),
    );
  }
}


```

and shows this when i print the error: `UnimplementedError: enablePersistence() is only available for Web. Use [Settings.persistenceEnabled] for other platforms.
`

### Expected behavior
I expected to work perfectly cause i create a new firebase and flutter project and i haven't any other complex code or anything. 

### Flutter doctor

<details><summary>Click To Expand</summary>

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.3.10, on macOS 11.7.1 20G918 darwin-x64, locale en-US)
[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 13.0)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2021.2)
[✓] VS Code (version 1.74.3)
[✓] Connected device (4 available)
[✓] HTTP Host Availability

• No issues found!
```

</details>

---

### Flutter dependencies

<details><summary>Click To Expand</summary>

```
Dart SDK 2.18.6
Flutter SDK 3.3.10
flutterfire 1.0.0+1

dependencies:
- cloud_firestore 4.3.1 [cloud_firestore_platform_interface cloud_firestore_web collection firebase_core firebase_core_platform_interface flutter meta]
- cupertino_icons 1.0.5
- firebase_auth 4.2.5 [firebase_auth_platform_interface firebase_auth_web firebase_core firebase_core_platform_interface flutter meta]
- firebase_core 2.4.1 [firebase_core_platform_interface firebase_core_web flutter meta]
- flutter 0.0.0 [characters collection material_color_utilities meta vector_math sky_engine]

dev dependencies:
- flutter_lints 2.0.1 [lints]
- flutter_test 0.0.0 [flutter test_api path fake_async clock stack_trace vector_math async boolean_selector characters collection matcher material_color_utilities meta source_span stream_channel string_scanner term_glyph]

transitive dependencies:
- _flutterfire_internals 1.0.12 [collection firebase_core firebase_core_platform_interface flutter meta]
- async 2.9.0 [collection meta]
- boolean_selector 2.1.0 [source_span string_scanner]
- characters 1.2.1
- clock 1.1.1
- cloud_firestore_platform_interface 5.10.1 [_flutterfire_internals collection firebase_core flutter meta plugin_platform_interface]
- cloud_firestore_web 3.2.1 [_flutterfire_internals cloud_firestore_platform_interface collection firebase_core firebase_core_web flutter flutter_web_plugins js]
- collection 1.16.0
- fake_async 1.3.1 [clock collection]
- firebase_auth_platform_interface 6.11.7 [_flutterfire_internals collection firebase_core flutter meta plugin_platform_interface]
- firebase_auth_web 5.2.4 [firebase_auth_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins http_parser intl js meta]
- firebase_core_platform_interface 4.5.2 [collection flutter flutter_test meta plugin_platform_interface]
- firebase_core_web 2.1.0 [firebase_core_platform_interface flutter flutter_web_plugins js meta]
- flutter_web_plugins 0.0.0 [flutter js characters collection material_color_utilities meta vector_math]
- http_parser 4.0.2 [collection source_span string_scanner typed_data]
- intl 0.17.0 [clock path]
- js 0.6.4
- lints 2.0.1
- matcher 0.12.12 [stack_trace]
- material_color_utilities 0.1.5
- meta 1.8.0
- path 1.8.2
- plugin_platform_interface 2.1.3 [meta]
- sky_engine 0.0.99
- source_span 1.9.0 [collection path term_glyph]
- stack_trace 1.10.0 [path]
- stream_channel 2.1.0 [async]
- string_scanner 1.1.1 [source_span]
- term_glyph 1.2.1
- test_api 0.4.12 [async boolean_selector collection meta source_span stack_trace stream_channel string_scanner term_glyph matcher]
- typed_data 1.3.1 [collection]
- vector_math 2.1.2
```

</details>

---

**Comment by darshankawar** [comment]
Thanks for the report @YawarOsman 
I ran the code which compiled and worked fine as expected without throwing the reported error.
When you say `firestore offline`, can you elaborate on it ? How are you simulating this scenario ?

**Comment by YawarOsman** [comment]
> Thanks for the report @YawarOsman I ran the code which compiled and worked fine as expected without throwing the reported error. When you say `firestore offline`, can you elaborate on it ? How are you simulating this scenario ?

for the current project it is fixed idk how, but i have another project that is a little bit big, and when i try to enable the persistence mode it show this error:

```
[cloud_firestore/failed-precondition] Firestore has already been started and persistence can no longer be enabled. You can only enable persistence before calling any other methods on a Firestore object.
```

and this how i did that in my main file:

```
void main() async {
  WidgetsFlutterBinding.ensureInitialized();

  await Firebase.initializeApp(
    options: DefaultFirebaseOptions.currentPlatform,
  );
   if(kIsWeb){
   try {
    await FirebaseFirestore.instance
        .enablePersistence(const PersistenceSettings(synchronizeTabs: true));
  } catch (e) {
    print(e);
  }
 }
}
```


And about your question i think offline mode is the way of using firebase offline read/ write data and store loaded data when the internet has been available

> Thanks for the report @YawarOsman I ran the code which compiled and worked fine as expected without throwing the reported error. When you say `firestore offline`, can you elaborate on it ? How are you simulating this scenario ?

**Comment by darshankawar** [comment]
> for the current project it is fixed idk how, but i have another project that is a little bit big, and when i try to enable the persistence mode it show this error:
> 
> ```
> [cloud_firestore/failed-precondition] Firestore has already been started and persistence can no longer be enab
> ```

I observed this error if I run the code sample you provided on web and try to hot restart. 
Also, if you see the error message, it indicates since the persistence is already enabled, we can't re-enable it and it can be instantiated only once, so I think in a nutshell, the error thrown is expected.

As you said in your another project, try to see if there's another instance of persistence being used which might be getting called, otherwise, I don't see such error getting triggered.

**Comment by YawarOsman** [comment]
> > for the current project it is fixed idk how, but i have another project that is a little bit big, and when i try to enable the persistence mode it show this error:
> > ```
> > [cloud_firestore/failed-precondition] Firestore has already been started and persistence can no longer be enab
> > ```
> 
> I observed this error if I run the code sample you provided on web and try to hot restart. Also, if you see the error message, it indicates since the persistence is already enabled, we can't re-enable it and it can be instantiated only once, so I think in a nutshell, the error thrown is expected.
> 
> As you said in your another project, try to see if there's another instance of persistence being used which might be getting called, otherwise, I don't see such error getting triggered.

I think it is because hot restart or something idk if i'm wrong or not.

anyway it work now but there is another issue here, when i try to fetch a document it shows this error:

> Failed to get document because the client is offline.

**Comment by danagbemava-nc** [comment]
Hi @YawarOsman, were you able to fetch some data before testing the offline mode? Because if you didn't then this is working as intended. See https://github.com/firebase/firebase-js-sdk/issues/3207#issuecomment-643810373

**Comment by YawarOsman** [comment]
> Hi @YawarOsman, were you able to fetch some data before testing the offline mode? Because if you didn't then this is working as intended. See [firebase/firebase-js-sdk#3207 (comment)](https://github.com/firebase/firebase-js-sdk/issues/3207#issuecomment-643810373)

The link he provided has been expired.


So it means the data that i want to fetch in the offline mode have to be uploaded since the offline mode enabled?
because i had some data in the firestore before enabling persistence mode, and i need those data when the application started running.

**Comment by darshankawar** [comment]
> anyway it work now but there is another issue here, when i try to fetch a document it shows this error:
> 
> > Failed to get document because the client is offline.

Please file a new issue by providing relevant details so that we can address it properly.
Closing this issue based on your above comment.
